google-analytics-query.html

```diff
@@ -208,105 +240,137 @@
-            if (this.output) query.output = this.output;
-            if (this.fields) query.fields = this.fields;
-
-            this.beforeQuerySend(query);
-
-            gapi.client.analytics.data.ga.get(query)
-                .execute(this.handleResponse.bind(this));
-
-            return true;
-          }
+        underscoreSrc: {
+          type: String,
+          value: 'gwc-ga-query'
         },
 
         /**
-         * The callback for the query run in `getData`. This is a separate
-         * function so subclasses can alter how the response is handled.
-         *
-         * @method handleResponse
+         * true if data is getting loeaded
+         * @attribute loading
+         * @type Boolean
          */
-        handleResponse: function(response) {
-          if (response.error) {
-            this.fire('analytics-query-error', response.error);
-          }
-          else {
-            this.data = response;
-            this.fire('analytics-query-success', response);
-          }
+        loading: {
+          type: Boolean,
+          value: false,
+          notify: true
         },
 
         /**
-         * A hook to alter the query params object (if needed) before
-         * the request is made. This method allows subclasses to change the
-         * request without overriding the entire `getData` method.
+         * 
          *
-         * @method beforeQuerySend
-         * @param {Object} query - The query params object.
+         * @attribute getDataResponseHandler
+         * @type Function
          */
-        beforeQuerySend: function(query) {
-          query._src = 'gwc-ga-query';
-        },
-
-        get hasRequiredParams() {
-          return !!(this.ids && this.metrics && this.startDate && this.endDate);
+        getDataResponseHandler: {
+          type: Function
         },
+        /**
+         * True if setup is ready 
+         * 
+         * @attribute setupReady
+         * @type Boolean
+         */
+        setupReady: {
+          type: Boolean,
+          observer: 'setupReadyChanged'
+        }
 
-      });
+      },
 
-    }());
+      observers: [
+        'getData(ids startDate endDate metrics dimensions sort filters segment samplingLevel startIndex maxResults output fields)'
+      ],
 
-  </script>
+      setupReadyChanged: function(newVal) {
+        if (newVal)
+          this.getData();
+        else
+          this.data = null;
+      },
+
+      /**
+       * Query the Google Analytics Core Reporting API.
+       *
+       * @method getData
+       */
+      getData: function() {
+        if (this.setupReady && this.hasRequiredParams) {
+          this.loading = true;
+          // Required parameters.
+          var query = {
+            'ids': this.ids,
+            'start-date': this.startDate,
+            'end-date': this.endDate,
+            'metrics': this.metrics,
+            '_src': this.underscoreSrc
```

Comment:
As mentioned above, the `'gwc-ga-query'` value can probably just be hardcoded here, or perhaps reference a constant declared at the top of the file.